core/src/test/java/bisq/core/btc/wallet/BsqWalletServiceTest.java

```diff
@@ -0,0 +1,43 @@
+package bisq.core.btc.wallet;
+
+import bisq.core.btc.exceptions.BsqChangeBelowDustException;
+import bisq.core.btc.exceptions.InsufficientBsqException;
+import bisq.core.btc.exceptions.TransactionVerificationException;
+import bisq.core.btc.exceptions.WalletException;
+import bisq.core.btc.setup.WalletsSetup;
+import bisq.core.dao.DaoKillSwitch;
+import bisq.core.dao.state.DaoStateService;
+import bisq.core.dao.state.unconfirmed.UnconfirmedBsqChangeOutputListService;
+import bisq.core.provider.fee.FeeService;
+import bisq.core.user.Preferences;
+import bisq.core.util.ParsingUtils;
+
+import org.bitcoinj.core.Coin;
+import org.bitcoinj.params.UnitTestParams;
+import org.bitcoinj.utils.MonetaryFormat;
+import org.bitcoinj.wallet.Wallet;
+
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BsqWalletServiceTest {
+    @Test(expected = InsufficientBsqException.class)
+    public void throwsExceptionWhenFundsAreInsufficient() throws TransactionVerificationException, WalletException, BsqChangeBelowDustException, InsufficientBsqException {
```

Comment:
Could you separate the code with `//Given`, `//When`, `//Then` comments?

Reply:
I don't really like that style and I think it adds nothing. actually any comment in a test is a code smell. 
Personally I use https://github.com/dmcg/minutest for my own work, thats nice and expressive.

Reply:
Well, to me this code is not readable and tests should be very obvious. I'm not sure which part is the setup and which one is the action.

Reply:
yes its not very readable, and its not a very useful test, the main motivation for this pr is to make the wallet service testable at all. 

All the test code is basically "given", the last line is "when", and the "then" is in the annotation at the top (we expect a exception).